bulk_indexer.go

```diff
@@ -138,14 +157,13 @@ func init() {
 
 // NewBulkIndexer returns a bulk indexer that issues bulk requests to Elasticsearch.
 // It is only tested with v8 go-elasticsearch client. Use other clients at your own risk.
-func NewBulkIndexer(client esapi.Transport, compressionLevel int, maxDocRetry int) *BulkIndexer {
+func NewBulkIndexer(cfg BulkIndexerConfig) *BulkIndexer {
```

Comment:
It just occured to me that we're not validating the config options anywhere. If `Client` is nil, it will panic. I think this was okay before, since the type was unexported, but since the type is not exported and we're "properly" supporting it, it'd be worthy to add config validation and perhaps returning an `error` in the constructor.

Previously, the options were validated by the Appender's config, but now this type may be used standalone. WDYT?

Reply:
good point. I've added nil client check to appender as well.